pkg/keybase/keybase.go

```diff
@@ -0,0 +1,152 @@
+)
+`
+
+func writeMirrorScript(configDir string) (out string, err error) {
+	script := []byte(mirrorScript)
+
+	scriptPath := filepath.Join(configDir, "keybase-mirror.cmd")
+	if _, err := os.Stat(scriptPath); os.IsNotExist(err) {
+		err := ioutil.WriteFile(scriptPath, script, 0700)
+		if err != nil {
+			return "", err
+		}
+	}
+	return scriptPath, nil
+}
+
+// Start keybase mirror [windows only]
+func StartMirror(configDir string, debug bool, logger logrus.Entry) error {
+
+	keybaseTestDir := filepath.Join(configDir, "keybase", "private")
+
+	// keybase mirror already started.  Nothing to do
+	if _, err := os.Stat(keybaseTestDir); err == nil {
+		logger.Info("Keybase mirror already started")
+		return nil
+	}
+	logger.Info("Writing keybase mirror script")
+	scriptPath, err := writeMirrorScript(configDir)
+	if err != nil {
+		return err
+	}
+	commandString := fmt.Sprintf("powershell %s %s", scriptPath, "start")
+	if debug {
+		logrus.Infof("Starting Keybase mirror with command; %s", commandString)
+	}
+	commandMap := strings.Split(commandString, " ")
+	if err := sh.Command(commandMap[0], commandMap[1:]).SetStdin(os.Stdin).Run(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Stop keybase mirror [windows only]
+func StopMirror(configDir string, debug bool, logger logrus.Entry) error {
+	// TODO(rluckie) Fix StopMirror to work with multiple KDK containers
+	// Use docker client to iterate though all running containers and ensure that no containers have mirror dir mounted
+	// If so, stop mirror
+	logger.Info("Writing keybase mirror script")
+	scriptPath, err := writeMirrorScript(configDir)
+	if err != nil {
+		return err
+	}
+	commandString := fmt.Sprintf("powershell %s %s", scriptPath, "stop")
+	if debug {
+		logrus.Infof("Stopping Keybase mirror with command; %s", commandString)
+	}
+	commandMap := strings.Split(commandString, " ")
+	if err := sh.Command(commandMap[0], commandMap[1:]).SetStdin(os.Stdin).Run(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Get keybase mounts
+// Linux & OSX: Detect /keybase
+// Windows10: Detect k: and /k
+func GetMounts(configRootDir string, logger logrus.Entry) (source string, target string, err error) {
+
+	keybaseRoots := []string{"/keybase", "k:", "/k"}
+	keybaseTestSubdir := "/private"
+	for _, keybaseRoot := range keybaseRoots {
+		if absPath, err := filepath.Abs(filepath.Join(keybaseRoot, keybaseTestSubdir)); err == nil {
+			if path, err := filepath.EvalSymlinks(absPath); err == nil {
+				source := filepath.Dir(path)
+				target := "/keybase"
+
+				logger.Infof("Detected keybase filesystem at: %v", source)
+
+				prmpt := prompt.Prompt{
+					Text:     "Mount your keybase directory within KDK? [y/n] ",
+					Loop:     true,
+					Validate: prompt.ValidateYorN,
+				}
+				if result, err := prmpt.Run(); err == nil && result == "y" {
+					logger.Info("Adding /keybase mount to configuration")
+					if runtime.GOOS == "windows" {
```

Comment:
With so many levels of nested conditions, it might be better to break out some of it into functions if not structs + methods.

(may)

Reply:
Agreed.  I'm going to leave this code as is, since it's got a lot of test mileage on it (and we don't have automated tests).  Bad on us.